lambdas/InspectMediaLambdaFunction/ebu-r-128.js

```diff
@@ -0,0 +1,85 @@
+const childProcess = require('child_process');
+const os = require('os');
+
+/**
+ * @typedef {object} FfmpegLoudnormResult
+ * @property {string} [input_i]
+ * @property {string} [input_tp]
+ * @property {string} [input_lra]
+ * @property {string} [input_thresh]
+ */
+
+module.exports = {
+  /**
+   * @param {string} filePath
+   * @returns {Promise<FfmpegLoudnormResult>}
+   */
+  inspect: async function inspect(filePath) {
+    return new Promise((resolve, reject) => {
+      const start = process.hrtime();
+
+      // This will return JSON data after a lot of other FFmpeg process
+      // information
+      const childProc = childProcess.spawn(
+        '/opt/bin/ffmpeg',
+        [
+          '-v',
+          'info',
+          '-hide_banner',
+          '-nostats',
+          '-i',
+          filePath,
+          '-af',
+          'loudnorm=dual_mono=true:print_format=json',
+          '-f',
+          'null',
+          '-',
+        ],
+        {
+          env: process.env,
+          cwd: os.tmpdir(),
+        },
+      );
+      const resultBuffers = [];
+
+      // The data we want seems to end up in stderr, so add that to the buffer
+      // capture as well
+      childProc.stdout.on('data', (buffer) => resultBuffers.push(buffer));
+      childProc.stderr.on('data', (buffer) => {
+        console.error(buffer.toString());
+        resultBuffers.push(buffer);
+      });
+
+      childProc.on('exit', (code, signal) => {
+        const end = process.hrtime(start);
+        console.log(
+          JSON.stringify({
+            msg: 'Finished ffmpeg loudnorm',
+            duration: `${end[0]} s ${end[1] / 1000000} ms`,
+          }),
+        );
+
+        if (code || signal) {
+          reject(new Error(`ffprobe failed with ${code || signal}`));
+        } else {
+          const output = Buffer.concat(resultBuffers).toString().trim();
+          // The JSON loudness data should come at the end of the output
+          const match = output.match(/\n({[\s\S]+})$/m);
```

Comment:
I could not come up with a version of the `ffmpeg` command that output JSON from the `loudnorm` filter, but **didn't** output anything else. So the buffer output always ends up with JSON _somewhere_ in there, and this is my attempt to get it out.

It seems like there should be a better way, but I couldn't find it.

Reply:
https://stackoverflow.com/questions/58994439/isolate-the-output-only-ffmpeg

Reply:
this probably isn't better - just moves the regex to the call